src/chainlock/signing.h

```diff
@@ -19,11 +20,25 @@ class CSigSharesManager;
 } // namespace llmq
 
 namespace chainlock {
-class ChainLockSigner : public llmq::CRecoveredSigsListener
+class ChainLockSignerParent
```

Comment:
`ChainLockSignerParent` is meant to be used as public interface of chainlock::

I'd suggest better name such as `ChainLockInterface` or `ChainLockManager`. And maybe rename `CChainLocksHandler` to `ChainLockImpl